tests/system-tests/internal/sniffer/sniffer.go

```diff
@@ -0,0 +1,633 @@
+	searchString string,
+	timeSpan time.Time) (int, error) {
+	if targetProtocol != httpProtocol && targetProtocol != udpProtocol {
+		return 0, fmt.Errorf("scanPacketSnifferDeploymentPodLogs supports only %s and %s protocols",
+			httpProtocol, udpProtocol)
+	}
+
+	snifferPodObjects, err := pod.List(apiClient, snifferNamespace,
+		metav1.ListOptions{LabelSelector: snifferSelectorLabel})
+	if err != nil {
+		return 0, err
+	}
+
+	matchesFound := 0
+
+	var snifferPodObj *pod.Builder
+
+	for _, podObj := range snifferPodObjects {
+		if podObj.Object.Spec.NodeName == proberPod.Object.Spec.NodeName {
+			glog.V(100).Infof("sniffer pod: %s", podObj.Definition.Name)
+			snifferPodObj = podObj
+		}
+	}
+
+	if snifferPodObj == nil {
+		return 0, fmt.Errorf("sniffer pod not found running on the %s node", proberPod.Object.Spec.NodeName)
+	}
+
+	logStartTimestamp := time.Since(timeSpan)
+	glog.V(100).Infof("\tTime duration is %s", logStartTimestamp)
+
+	if logStartTimestamp.Abs().Seconds() < 1 {
+		logStartTimestamp, err = time.ParseDuration("1s")
+		if err != nil {
+			return 0, fmt.Errorf("failed to parse time duration: %w", err)
+		}
+	}
+
+	logs, err := snifferPodObj.GetLog(logStartTimestamp, snifferPodObj.Definition.Spec.Containers[0].Name)
+
+	glog.V(100).Infof("debug log for %s: %s", snifferPodObj.Definition.Name, logs)
+
+	if err != nil {
+		glog.V(100).Infof("Failed to get logs from pod %q: %v",
+			snifferPodObj.Definition.Name, err)
+
+		return 0, fmt.Errorf("failed to get logs from pod %q: %w",
+			snifferPodObj.Definition.Name, err)
+	}
+
+	if len(logs) != 0 {
+		buf := new(bytes.Buffer)
+		_, err = buf.WriteString(logs)
+
+		if err != nil {
+			return 0, fmt.Errorf("error in copying info from pod logs to buffer")
+		}
+
+		_ = buf.String()
+
+		scanner := bufio.NewScanner(buf)
+		for scanner.Scan() {
+			logLine := scanner.Text()
+			if strings.Contains(logLine, searchString) {
+				glog.V(100).Infof("Found hit in log line: %s", logLine)
+				logLineExploded := strings.Fields(logLine)
+
+				if len(logLineExploded) != 2 {
+					return 0, fmt.Errorf("unexpected logline content: %s", logLine)
+				}
+
+				matchesFound++
+			}
+		}
+
+		return matchesFound, nil
+	}
+
+	return 0, fmt.Errorf("no matches found in sniffer log")
+}
+
+// sendProbesToHostPort runs curl against a http://targetHost:targetPort for the specified number of iterations.
+// Returns the "targetHost:targetPort" string that will be used for the requests search.
+func sendProbesToHostPort(
+	proberPod *pod.Builder,
+	targetProtocol,
+	targetHost string,
+	targetPort,
+	iterations int) (string, error) {
+	if proberPod == nil {
+		return "", fmt.Errorf("nil pointer to proberPod was provided in sendProbesToHostPort")
+	}
+
+	if targetProtocol != httpProtocol && targetProtocol != udpProtocol {
+		return "", fmt.Errorf("sendProbesToHostPort supports only %s and %s protocols",
+			httpProtocol, udpProtocol)
+	}
+
+	targetURL := fmt.Sprintf("%s:%d", targetHost, targetPort)
```

Comment:
U might need extra check for the IPv6 address as it should be wrapped with square brackets